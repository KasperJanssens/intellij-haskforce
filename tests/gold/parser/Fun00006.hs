module Fun00006 where

rev :: [a] -> [a]
rev xs = go [] xs
    where go acc []     = acc
          go acc (x:xs) = go (x:acc) xs
