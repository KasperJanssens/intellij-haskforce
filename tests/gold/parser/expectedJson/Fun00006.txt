[{"Module":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":7,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":1}]},{"ModuleHead":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":22,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":22,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":17}]},{"ModuleName":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":16,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":8},"srcInfoPoints":[]},"Fun00006"]},null,null]},[],[],[{"TypeSig":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":5}]},[{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":4,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},"srcInfoPoints":[]},"rev"]}],{"TyFun":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":8},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":14,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":12}]},{"TyList":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":11,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":8},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":8},{"srcSpanFilename":"A.hs","srcSpanEndColumn":11,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":10}]},{"TyVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":10,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":9},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":10,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":9},"srcInfoPoints":[]},"a"]}]}]},{"TyList":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":15},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":16,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":15},{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":17}]},{"TyVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":16},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":16},"srcInfoPoints":[]},"a"]}]}]}]}]},{"FunBind":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":4,"srcSpanEndLine":7,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":10,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":5}]},[{"Match":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":4,"srcSpanEndLine":7,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":10,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":5}]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":4,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":1},"srcInfoPoints":[]},"rev"]},[{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":5},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":5},"srcInfoPoints":[]},"xs"]}]}],{"UnGuardedRhs":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":8},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":8}]},{"App":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":10},"srcInfoPoints":[]},{"App":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":15,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":10},"srcInfoPoints":[]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":12,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":10},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":12,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":10},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":12,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":10},"srcInfoPoints":[]},"go"]}]}]},{"List":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":15,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":13},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":14,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":13},{"srcSpanFilename":"A.hs","srcSpanEndColumn":15,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":14}]},[]]}]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":16},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":16},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":16},"srcInfoPoints":[]},"xs"]}]}]}]}]},{"BDecls":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":5,"srcSpanEndLine":7,"srcSpanStartColumn":11},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":11,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":11},{"srcSpanFilename":"A.hs","srcSpanEndColumn":11,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":11},{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":1}]},[{"FunBind":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":40,"srcSpanStartLine":5,"srcSpanEndLine":6,"srcSpanStartColumn":11},"srcInfoPoints":[]},[{"Match":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":30,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":11},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":13,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":11},"srcInfoPoints":[]},"go"]},[{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":14},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":14},"srcInfoPoints":[]},"acc"]}]},{"PList":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":20,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":18},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":18},{"srcSpanFilename":"A.hs","srcSpanEndColumn":20,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":19}]},[]]}],{"UnGuardedRhs":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":30,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":25},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":26,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":25}]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":30,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":27},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":30,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":27},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":30,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":27},"srcInfoPoints":[]},"acc"]}]}]}]},null]},{"Match":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":40,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":11},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":13,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":11},"srcInfoPoints":[]},"go"]},[{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":14},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":14},"srcInfoPoints":[]},"acc"]}]},{"PParen":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":24,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":18},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":18},{"srcSpanFilename":"A.hs","srcSpanEndColumn":24,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":23}]},{"PInfixApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":23,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":19},"srcInfoPoints":[]},{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":20,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":19},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":20,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":19},"srcInfoPoints":[]},"x"]}]},{"Special":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":20},"srcInfoPoints":[]},{"Cons":{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":20},"srcInfoPoints":[]}}]},{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":23,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":21},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":23,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":21},"srcInfoPoints":[]},"xs"]}]}]}]}],{"UnGuardedRhs":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":40,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":25},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":26,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":25}]},{"App":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":40,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":27},"srcInfoPoints":[]},{"App":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":37,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":27},"srcInfoPoints":[]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":29,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":27},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":29,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":27},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":29,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":27},"srcInfoPoints":[]},"go"]}]}]},{"Paren":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":37,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":30},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":31,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":30},{"srcSpanFilename":"A.hs","srcSpanEndColumn":37,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":36}]},{"InfixApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":36,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":31},"srcInfoPoints":[]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":32,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":31},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":32,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":31},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":32,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":31},"srcInfoPoints":[]},"x"]}]}]},{"QConOp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":33,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":32},"srcInfoPoints":[]},{"Special":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":33,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":32},"srcInfoPoints":[]},{"Cons":{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":33,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":32},"srcInfoPoints":[]}}]}]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":36,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":33},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":36,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":33},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":36,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":33},"srcInfoPoints":[]},"acc"]}]}]}]}]}]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":40,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":38},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":40,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":38},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":40,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":38},"srcInfoPoints":[]},"xs"]}]}]}]}]},null]}]]}]]}]}]]}]]},[]]