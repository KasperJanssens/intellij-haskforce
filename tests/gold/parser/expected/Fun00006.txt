Haskell File
  PsiElement(HaskellTokenType.module)('module')
  HaskellQconidImpl(QCONID)
    PsiElement(HaskellTokenType.conid)('Fun00006')
  PsiElement(HaskellTokenType.where)('where')
  HaskellBodyImpl(BODY)
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('rev')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.::)('::')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.[)('[')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('a')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.])(']')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.->)('->')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.[)('[')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('a')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.])(']')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('rev')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('xs')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.=)('=')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('go')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.[)('[')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.])(']')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('xs')
    PsiElement(HaskellTokenType.where)('where')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('go')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('acc')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.[)('[')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.])(']')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.=)('=')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('acc')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('go')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('acc')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.()('(')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('x')
    HaskellQconsymImpl(QCONSYM)
      HaskellConsymImpl(CONSYM)
        PsiElement(HaskellTokenType.:)(':')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('xs')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.))(')')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.=)('=')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('go')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.()('(')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('x')
    HaskellQconsymImpl(QCONSYM)
      HaskellConsymImpl(CONSYM)
        PsiElement(HaskellTokenType.:)(':')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('acc')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.))(')')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('xs')